### ISSUE TITLE
[BUG] Protonvpn-cli 2.2.4 doesn't work with ([!] There was an error connecting to the ProtonVPN API) ERROR

### ISSUE THREAD

**Issue by farzinh** [ProtonVPN/linux-cli#235]
Protonvpn-cli version 2.2.4 doesn't work on Linux with this errors:

[!] There was an error connecting to the ProtonVPN API.
[!] Please make sure your connection is working properly!

after searching around I find out error comes from this version 2.2.4, now I just downgrade to 2.1.0 and everything works properly.

I checked out utils.py file in both of versions and It seems like problem comes from API initiation, protonvpn.com is a filtered domain in countries like Iran but in 2.1.0 API comes from protonmail.com and it's working properly. 

**Desktop (please complete the following information):**
 - OS: MX Linux 19.2
 - Python Version: python3.7
 - ProtonVPN-CLI Version: 2.2.4

**Comment by ghonchesefidi** [comment]
Check this link:
https://github.com/ProtonVPN/linux-cli/issues/97

**Comment by aliazani** [comment]
@Rafficer I have the same problem please check the solution here it make sense I think:
https://github.com/ProtonVPN/linux-cli/pull/237

**Comment by Rafficer** [comment]
No need for a proxy, you can change the api domain in the config file.

**Comment by aliazani** [comment]
@Rafficer I changed api_domain to https://api.protonmail.ch/ but still have the same problem

**Comment by Rafficer** [comment]
It's just supposed to be `api.protonmail.ch` iirc.

**Comment by aliazani** [comment]
@Rafficer Thanks I changed it and now it works properly :)

**Comment by iiriix** [comment]
> No need for a proxy, you can change the api domain in the config file.

@Rafficer This can be easily blocked again. Isn't it better to add proxy support for the users who need it?

**Comment by Rafficer** [comment]
Brings the issue of "do you trust the proxy". The better solution is to implement alternative routing, which is being worked on.

**Comment by iiriix** [comment]
This can be an optional feature for those who "trust their proxy", and of course, it will be disabled by default. :-/

**Comment by molaeiali** [comment]
@Rafficer "Proxy" is not necessarily a server which we have no control on, It can be Tor, or a ssh tunnel to our own server, etc. Clearly, the proxy is just a option that can be used by users who live in countries with blocked ProtonVPN domains.

**Comment by Rafficer** [comment]
Definitely, but it also gives users that don't know any better the possibility to reveal information they shouldn't reveal.

**Comment by iiriix** [comment]
With all due respect, it doesn't make sence. It's similar to "Custom DNS", "Kill Switch", or even "Split Tunneling". If you don't know what you're doing, all of them can reveal information you shouldn't reveal. :)

**Comment by jdbancal** [comment]
Same problem here, but from France(!). Also, `ping api.protonvpn.ch` is successful. Nevertheless, the command `sudo protonvpn init` returns
```
[!] There was an error connecting to the ProtonVPN API.
[!] Please make sure your connection is working properly!
```

After changing the api server to `api.protonmail.ch` in `/usr/local/lib/python3.8/dist-packages/protonvpn_cli/utils.py`, `sudo protonvpn init` is successful. However, I still cannot connect, the command `sudo protonvpn c` ends with the following message:
```
Connecting to CH#18-TOR via TCP...
[!] Authentication failed. 
[!] Please make sure that your Username and Password is correct.
```
Using here the password with which I can log onto `http://www.protonvpn.com`. Hence, I still cannot connect from my PC.

**Comment by jdbancal** [comment]
All the docs out there may not be consistent at the moment... Maybe this could be linked with using the protonvpn login credentials, instead of the openvpn ones (accessible through the settings menu on the protonvpn website)

**Comment by farzinh** [comment]
I'm using version 2.1.0, and it's working like a charm.

**Comment by JCDetwiler** [comment]
OP's original post helped me; I'm on Manjaro currently. I was experiencing the

    [!] There was an error connecting to the ProtonVPN API.
    [!] Please make sure your connection is working properly!

message from my new house where I had new service (pretty crappy) and a new router. If I ran the `status` command after connecting, I would find that my previous connection had resumed. However, everything was working whenever I would connect elsewhere, like on campus at my university, making me think it had to be one of those two causing the problem.

I just ran `sudo protonvpn init` again and that fixed it. Hopefully it'll stay fixed.

**Comment by kopach** [comment]
hi there. any updates on this. I'm facing the same issue

**Comment by JCDetwiler** [comment]
> hi there. any updates on this? I'm facing the same issue

There's a new app altogether now.
https://protonvpn.com/blog/protonvpn-linux-app/

**Comment by JCDetwiler** [comment]
> > There's a new app altogether now. [protonvpn.com/blog/protonvpn-linux-app](https://protonvpn.com/blog/protonvpn-linux-app/)
> 
> It seems to not support CLI mode and requires GUI (which I do not have on my machine)

You're looking for this then. https://protonvpn.com/support/linux-vpn-tool/

**Comment by em20nn** [comment]
I don't know what caused my error but I could overcome with this line;
`sudo protonvpn c -f  (for fastest) -p tcp (or udp)`
I dont know what the issue is and how to fix, but this line made me connect to the fastest server using tcp protocol.

**Comment by Alireza17224** [comment]
Same issue

**Comment by jh-devv** [comment]
Same issue on NIX OS with the CLI and GUI

**Comment by PAVAN1698** [comment]
I am getting the below error while using Proton VPN CLI.
Is this information correct? [Y/n]: Y
Writing configuration to disk...
[!] There was an error with accessing the ProtonVPN API.
[!] Please make sure your connection is working properly!
[!] HTTP Error Code: 422

**Comment by prasanshivarapu** [comment]
> I am getting the below error while using Proton VPN CLI. Is this information correct? [Y/n]: Y Writing configuration to disk... [!] There was an error with accessing the ProtonVPN API. [!] Please make sure your connection is working properly! [!] HTTP Error Code: 422

same im also getting, 
did u find any solution

**Comment by kirarpit** [comment]
Same issue on Ubuntu 24.04 headless server. 
Had to install venv and cd to venv's bin to run the following command:
$ sudo ./python3 ./protonvpn init

which gives the same error as mentioned by others:
Writing configuration to disk...
[!] There was an error with accessing the ProtonVPN API.
[!] Please make sure your connection is working properly!
[!] HTTP Error Code: 422

**Comment by GitHubQueenn** [comment]
> Writing configuration to disk... [!] There was an error with accessing the ProtonVPN API. [!] Please make sure your connection is working properly! [!] HTTP Error Code: 422

Same issue. No reply here since September?....Not a good look

**Comment by PIKUT0** [comment]
Yo, same issue here. API doesn't work properly on some of my devices
